src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -139,6 +139,121 @@ ASR::Module_t* load_module(Allocator &al, SymbolTable *symtab,
     return mod2;
 }
 
+bool types_equal(const ASR::ttype_t &a, const ASR::ttype_t &b) {
```

Comment:
Use `ASRUtils::check_equal_type()`, as in:
```c++
        if (ASRUtils::check_equal_type(ASRUtils::expr_type(target),
                                    ASRUtils::expr_type(value)))`
```
If this `types_equal` does anything else, then let's improve the `ASRUtils::check_equal_type` function.